src/aiida_quantumespresso/workflows/xspectra/crystal.py

```diff
@@ -0,0 +1,676 @@
+            core_wfc_data = shelljob_node.outputs.stdout
+            header_line = core_wfc_data.get_content()[:40]
+            num_core_states = int(header_line.split(' ')[5])
+            if num_core_states == 0:
+                return self.exit_codes.ERROR_NO_GIPAW_INFO_FOUND
+
+    def run_all_xspectra_core(self):
+        """Call all ``XspectraCoreWorkChain``s required to compute all requested spectra."""
+
+        structures_to_process = self.ctx.structures_to_process
+        equivalent_sites_data = self.ctx.equivalent_sites_data
+        abs_atom_marker = self.inputs.abs_atom_marker.value
+
+        for site in structures_to_process:
+            inputs = AttributeDict(self.exposed_inputs(XspectraCoreWorkChain, namespace='core'))
+            structure =structures_to_process[site]
+            inputs.structure = structure
+            abs_element = equivalent_sites_data[site]['symbol']
+
+            if 'core_hole_treatments' in self.inputs:
+                ch_treatments = self.inputs.core_hole_treatments.get_dict()
+                ch_treatment = ch_treatments.get(abs_element, 'full')
+            else:
+                ch_treatment = 'full'
+
+            inputs.metadata.call_link_label = f'{site}_xspectra'
+            inputs.eps_vectors = orm.List(list=self.ctx.eps_vectors)
+            if 'core_wfc_data' in self.inputs:
+                inputs.core_wfc_data = self.inputs.core_wfc_data[abs_element]
+            else:
+                inputs.core_wfc_data = self.ctx[f'upf2plotcore_{abs_element}'].outputs.stdout
+
+            # Get the given settings for the SCF inputs and then overwrite them with the
+            # chosen core-hole approximation, then apply the correct pseudopotential pair
+            scf_params = inputs.scf.pw.parameters.get_dict()
+            ch_inputs = XspectraCoreWorkChain.get_treatment_inputs(treatment=ch_treatment)
+            new_scf_params = recursive_merge(left=scf_params, right=ch_inputs)
+
+            # Set the parameter `xiabs` correctly, now that we know which structure we have
+            new_xs_params = inputs.xs_prod.xspectra.parameters.get_dict()
+            kinds_present = sorted([kind.name for kind in structure.kinds])
+            abs_species_index = kinds_present.index(abs_atom_marker) + 1
+            new_xs_params['INPUT_XSPECTRA']['xiabs'] = abs_species_index
+
+            # Set `starting_magnetization` if we are using an XCH approximation, using the
+            # absorbing species as a reasonable place for the unpaired electron.
+            if ch_treatment == 'xch_smear':
+                new_scf_params['SYSTEM'][f'starting_magnetization({abs_species_index})'] = 1
+
+            core_hole_pseudo = self.inputs.core_hole_pseudos[abs_element]
+            gipaw_pseudo = self.inputs.gipaw_pseudos[abs_element]
+            inputs.scf.pw.pseudos[abs_atom_marker] = core_hole_pseudo
+            inputs.scf.pw.pseudos[abs_element] = gipaw_pseudo
```

Comment:
If a user has a larger cell, so no supercell is created. For a special element, there is only one atom. In the default `inputs.scf.pw`, there is a pseudo for this element. Now this atom is replaced by the marker. So there will be pseudo corresponding to no kind anymore. This will raise an error.

Reply:
I will test and add a method to catch this, thus only setting `core_hole_pseudo` and not `gipaw_pseudo` for the element if there is only one instance of it in the structure.